app/Graph/build.cpp

```diff
@@ -200,4 +193,1114 @@ void build_graph(it_lab_ai::Tensor& input, it_lab_ai::Tensor& output,
+                      << " ===" << std::endl;
+                  std::cout << "Expected inputs: ";
+                  for (const auto& inp : concat_connections[target_name]) {
+                    std::cout << inp << " ";
+                  }
+                  std::cout << std::endl;
+
+                  std::cout << "Actual order: ";
+                  for (size_t i = 0; i < concat_orders[target_name].size();
+                       ++i) {
+                    std::cout << concat_orders[target_name][i];
+                    if (i < concat_orders[target_name].size() - 1)
+                      std::cout << ", ";
+                  }
+                  std::cout << std::endl;
+                }*/
+              }
+            }
+          }
+        }
+      }
+
+      try {
+        graph.makeConnection(*name_to_layer[source_name],
+                             *name_to_layer[target_name]);
+
+      } catch (const std::exception& e) {
+        std::cerr << "Failed: " << source_name << " -> " << target_name << " : "
+                  << e.what() << std::endl;
+      }
+    }
+  }
+  for (auto& split_dist : split_distribution) {
+    for (auto& connection : split_dist) {
+      for (const auto& [name, layer] : name_to_layer) {
+        if (original_ids[name] == connection.first) {
+          connection.first = layer->getID();
+          break;
+        }
+      }
+    }
+  }
+  graph.setSplitDistribution(split_distribution);
+  auto output_layer = layers.back();
+  graph.setOutput(*output_layer, output);
+  auto in_out_degrees = graph.getInOutDegrees();
+  auto traversal_order = graph.getTraversalOrder();
+
+  if (comments) std::cout << "Starting inference..." << std::endl;
+  try {
+    graph.inference();
+    if (comments) std::cout << "Inference completed successfully." << std::endl;
+  } catch (const std::exception& e) {
+    std::cerr << "ERROR during inference: " << e.what() << std::endl;
+  }
+
+#ifdef ENABLE_STATISTIC_TIME
+  std::vector<std::string> times = graph.getTimeInfo();
+  std::cout << "!INFERENCE TIME INFO START!" << std::endl;
+  for (size_t i = 0; i < times.size(); i++) {
+    std::cout << times[i] << std::endl;
+  }
+  std::vector<int> elps_time = graph.getTime();
+  int sum = std::accumulate(elps_time.begin(), elps_time.end(), 0);
+  std::cout << "Elapsed inference time:" << sum << std::endl;
+  std::cout << "!INFERENCE TIME INFO END!" << std::endl;
+#endif
+}
+
+std::unordered_map<int, std::string> load_class_names(
+    const std::string& filename) {
+  std::unordered_map<int, std::string> class_names;
+  std::ifstream file(filename);
+  std::string line;
+
+  if (!file.is_open()) {
+    throw std::runtime_error("Cannot open class names file: " + filename);
+  }
+
+  while (std::getline(file, line)) {
+    line = std::regex_replace(line, std::regex("^\\s+|\\s+$"), "");
+    if (line.empty()) continue;
+
+    std::regex pattern("(\\d+):\\s*'([^']+)'");
```

Comment:
Compiling `std::regex` inside a per-line loop in `load_class_names`. Move the regex creation outside the loop.